Refactors how floors turfs set broken/burnt states + fixes glass plating turning invisible when damaged

AlinaStarkova opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

adds sprites for both normal and reinforced glass plating (i reduced it to 2, from 5, from when the image was taken), the sprites arent the absolute best but arent terrible imho, 

refactor to floor code to *not* dumbly hardcode an initialization for broken_state list sprites, as well as moves burnt and broken states to procs, and has an override to all floor turfs which have both a broken and/or burnt icons unique to it

also changed how certain tiles react when broken, now, instead of magically turning into iron plating when stuff like dirt, lavaland basalt, sand, water, wasteland tiles, you name it. when they break, they will just use their basic icon (typically, wont change), rather than turn into iron plating magically. when asteroid turfs (lavaland basalt, icebox snow, asteroid sand etc..) which have a dug icon break, they change to their dug version instead


https://user-images.githubusercontent.com/40489693/104960109-963b2f80-59a1-11eb-8eb2-aca72b9207fe.mp4

(i know some people will prefer the icon to just not change when broken, but the snow version already goes to the dig icon, also you cant dig broken tiles, so its easier to show the dug icon so people know you cant dig that tile anymore)

removed dumb initialization override for plating

spriters can easily replace the sprites with something better if they can make any
![image](https://user-images.githubusercontent.com/40489693/104808945-84963400-57b7-11eb-9700-02bcd9f3e499.png)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

having plating turn invisible when damaged due to an oversight of the code is bad, e

(you cant actually repair glass plating, but atleast being able to SEE that its damaged is better than it just being invisible but there)

better code
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Nari Harimoto
fix: fixed glass plating turning invisible when damaged
refactor: changed some floor tile broken_state code to use a proc in initialization rather than hardcoded in
refactor: removed initialization override for plating.dm
fix: all non-iron based tiles will no longer magically turn into iron plating when broken, instead asteroid turf will become dug and stuff like grass, dirt, water etc. will not change icon when broken
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AlinaStarkova committed:
Adds damage sprites for glass plating

e -Narii

Tlaltecuhtli commented:
does it have burnt_states?

AlinaStarkova commented:
> 
> 
> does it have burnt_states?

burn-state in its parent path doesnt actually initialize its own list, so as far as i can tell, glass plating does not have any issues when being burnt, because initialization isnt setting any burn states, unlike damage states

Rohesie commented on code/game/turfs/open/glass.dm:
```diff
@@ -13,8 +13,11 @@
 	barefootstep = FOOTSTEP_HARD_BAREFOOT
 	clawfootstep = FOOTSTEP_HARD_CLAW
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	var/stateDamage = "glass"
```
```suggestion
	var/state_damage = "glass"
```
Favor snake case and please document new variables.

AlinaStarkova replied:
snake case?

Rohesie replied:
`snake_case`, `camelCase`, `PascalCase`

Rohesie commented on code/game/turfs/open/glass.dm:
```diff
@@ -13,8 +13,11 @@
+	var/stateDamage = "glass"
 
 /turf/open/floor/glass/Initialize()
+	if (!broken_states)
+		broken_states = string_list(list("[stateDamage]-damaged1", "[stateDamage]-damaged2", "[stateDamage]-damaged3", "[stateDamage]-damaged4", "[stateDamage]-damaged5"))
```
Does this have to be hardcoded on `Initialize()`?
I'd prefer this was manually setup on the definition, chiefly because this is not very expansible.

AlinaStarkova replied:
i believe so, the initialize in the basetype/parent creates that list directly in its initialization, so i need to override its initialization

AlinaStarkova committed:
Refactor floor broken_states

AlinaStarkova committed:
refactor some floor break code

changed it so it doesnt force initialize a broken_state list in floor initialization, rather it calls a proc that does that

removed the unnesscary initialize override for plating and set their states directly

reduced broken states for glass plating by 2, a couple of them looked too simialr to be worth it

removed unnesscary proc overide for glass platings

Rohesie commented on code/game/turfs/open/floor.dm:
```diff
@@ -44,6 +44,9 @@
 
+/turf/open/floor/proc/setup_broken_states()
+	broken_states = string_list(list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5"))
+	return
```
```suggestion

```
No need for empty returns at the end of a proc.

Rohesie commented on code/game/turfs/open/floor/plating.dm:
```diff
@@ -17,6 +17,8 @@
 	clawfootstep = FOOTSTEP_HARD_CLAW
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	burnt_states = list("panelscorched")
+	broken_states = list("platingdmg1", "platingdmg2", "platingdmg3")
```
Put this into `/turf/open/floor/plating/setup_broken_states()`

Rohesie commented on code/game/turfs/open/glass.dm:
```diff
@@ -13,6 +13,7 @@
 	barefootstep = FOOTSTEP_HARD_BAREFOOT
 	clawfootstep = FOOTSTEP_HARD_CLAW
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	broken_states = list("glass-damaged1", "glass-damaged2", "glass-damaged3")
```
And this into `/turf/open/floor/glass/setup_broken_states()`

Rohesie commented on code/game/turfs/open/glass.dm:
```diff
@@ -23,9 +24,11 @@
 	icon = 'icons/turf/floors/reinf_glass.dmi'
 	icon_state = "reinf_glass-0"
 	base_icon_state = "reinf_glass"
+	broken_states = list("reinf_glass-damaged1", "reinf_glass-damaged2", "reinf_glass-damaged3")
```
And this into `And this into `/turf/open/floor/glass/reinforced/setup_broken_states()``

AlinaStarkova committed:
refactors floor damage icons

changes how floors create their list of damage icons, changes how non-iron based floors react when damage (they no longer magically become iron plating)

Rohesie commented on code/game/turfs/open/floor/fancy_floor.dm:
```diff
@@ -11,13 +11,15 @@
 	tiled_dirt = FALSE
 
+/turf/open/floor/wood/setup_broken_states()
+	broken_states = list("wood-broken", "wood-broken2", "wood-broken3", "wood-broken4", "wood-broken5", "wood-broken6", "wood-broken7")
```
Either make them all stringlists on each proc redefinition, or make something like this on Initialize:

```
if (broken_states)
	stack_trace("the message I showed you before on discord")
else
	var/list/new_broken_states = setup_broken_states()
	if(new_broken_states)
		broken_states = string_list(new_broken_states)
```

Rohesie replied:
```suggestion
	return list("wood-broken", "wood-broken2", "wood-broken3", "wood-broken4", "wood-broken5", "wood-broken6", "wood-broken7")
```

AlinaStarkova committed:
fixed a runtime

Rohesie commented on code/game/turfs/open/floor.dm:
```diff
@@ -28,12 +28,18 @@
-		broken_states = string_list(list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5"))
+	if (broken_states)
+		stack_trace("broken_states defined at the object level for [type], move it to setup_broken_states()")
 	else
-		broken_states = string_list(broken_states)
-	if(burnt_states)
-		burnt_states = string_list(burnt_states)
+		var/list/new_broken_states = setup_broken_states()
+		if(new_broken_states)
+			broken_states = string_list(new_broken_states)
+	if (burnt_states)
+		stack_trace("burnt_states defined at the object level for [type], move it to setup_burnt_states()")
+	else
+		var/list/new_burnt_states = setup_burnt_states()
+		if(new_burnt_states)
+			burnt_states = string_list(new_burnt_states)
```
Call parent (. = ..) before these, instead of after.
At the start of the proc, if possible.
Does that break anything?

Rohesie commented on code/game/turfs/open/floor.dm:
```diff
@@ -44,6 +50,11 @@
 		GLOB.station_turfs += src
 
+/turf/open/floor/proc/setup_broken_states()
+	broken_states = string_list(list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5"))
```
```suggestion
	broken_states = list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
```
This will already be turned into a stringlist on initialization.

Rohesie commented on code/game/turfs/open/floor.dm:
```diff
@@ -28,12 +28,18 @@
-		broken_states = string_list(broken_states)
-	if(burnt_states)
-		burnt_states = string_list(burnt_states)
+		var/list/new_broken_states = setup_broken_states()
+		if(new_broken_states)
+			broken_states = string_list(new_broken_states)
```
```suggestion
		broken_states = string_list(setup_broken_states())
```
There should always be `broken_states`, unlike `burnt_states`, right?

AlinaStarkova replied:
yes

Rohesie commented on code/game/turfs/open/floor/fancy_floor.dm:
```diff
@@ -82,6 +83,9 @@
 	tiled_dirt = FALSE
 
+/turf/open/floor/grass/setup_broken_states()
+	broken_states = list("sand")
```
```suggestion
	return list("sand")
```

Rohesie commented on code/game/turfs/open/floor/fancy_floor.dm:
```diff
@@ -128,6 +131,9 @@
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
 
+/turf/open/floor/grass/snow/setup_broken_states()
+	broken_states = list("snow_dug")
```
```suggestion
	return list("snow_dug")
```

Rohesie commented on code/game/turfs/open/floor/fancy_floor.dm:
```diff
@@ -194,6 +199,9 @@
 	tiled_dirt = FALSE
 
+/turf/open/floor/carpet/setup_broken_states()
+	broken_states = list("damaged")
```
```suggestion
	return list("damaged")
```

Rohesie commented on code/game/turfs/open/floor/fancy_floor.dm:
```diff
@@ -388,10 +396,12 @@
 	tiled_dirt = FALSE
 
+/turf/open/floor/fakespace/setup_broken_states()
+	broken_states = list("damaged")
```
```suggestion
	return list("damaged")
```

Rohesie commented on code/game/turfs/open/floor/light_floor.dm:
```diff
@@ -24,6 +23,9 @@
 	var/static/list/lighttile_designs
 
+/turf/open/floor/light/setup_broken_states()
+	broken_states = list("light_broken")
```
```suggestion
	return list("light_broken")
```

Rohesie commented on code/game/turfs/open/floor/mineral_floor.dm:
```diff
@@ -17,11 +17,12 @@
 	icons = typelist("icons", icons)
 
+/turf/open/floor/mineral/setup_broken_states()
+	if(!broken_states)
+		broken_states = list("[initial(icon_state)]_dam")
```
```suggestion
	return list("[initial(icon_state)]_dam")
```

Rohesie commented on code/game/turfs/open/floor/mineral_floor.dm:
```diff
@@ -87,6 +88,8 @@
 	floor_tile = /obj/item/stack/tile/mineral/titanium
+
+/turf/open/floor/mineral/titanium/setup_broken_states()
 	broken_states = list("titanium_dam1","titanium_dam2","titanium_dam3","titanium_dam4","titanium_dam5")
```
```suggestion
	return list("titanium_dam1", "titanium_dam2", "titanium_dam3", "titanium_dam4", "titanium_dam5")
```

Rohesie commented on code/game/turfs/open/floor/mineral_floor.dm:
```diff
@@ -128,6 +131,8 @@
 	floor_tile = /obj/item/stack/tile/mineral/titanium/tiled
+
+/turf/open/floor/mineral/titanium/tiled/setup_broken_states()
 	broken_states = list("titanium_dam1_old","titanium_dam2_old","titanium_dam3_old","titanium_dam4_old","titanium_dam5_old")
```
```suggestion
	return list("titanium_dam1_old", "titanium_dam2_old", "titanium_dam3_old", "titanium_dam4_old", "titanium_dam5_old")
```

Rohesie commented on code/game/turfs/open/floor/mineral_floor.dm:
```diff
@@ -166,6 +171,8 @@
 	floor_tile = /obj/item/stack/tile/mineral/plastitanium
+
+/turf/open/floor/mineral/plastitanium/setup_broken_states()
 	broken_states = list("plastitanium_dam1","plastitanium_dam2","plastitanium_dam3","plastitanium_dam4","plastitanium_dam5")
```
```suggestion
	return list("plastitanium_dam1", "plastitanium_dam2", "plastitanium_dam3", "plastitanium_dam4", "plastitanium_dam5")
```

Rohesie commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -119,9 +119,13 @@
+	slowdown = -0.3
+
+/turf/open/floor/noslip/setup_broken_states()
 	broken_states = list("noslip-damaged1","noslip-damaged2","noslip-damaged3")
```
```suggestion
	return list("noslip-damaged1", "noslip-damaged2", "noslip-damaged3")
```

Rohesie commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -119,9 +119,13 @@
 	broken_states = list("noslip-damaged1","noslip-damaged2","noslip-damaged3")
+
+/turf/open/floor/noslip/setup_burnt_states()
 	burnt_states = list("noslip-scorched1","noslip-scorched2")
```
```suggestion
	return list("noslip-scorched1", "noslip-scorched2")
```

Rohesie commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -177,13 +181,17 @@
 	floor_tile = /obj/item/stack/tile/plastic
+
+/turf/open/floor/plastic/setup_broken_states()
 	broken_states = list("plastic-damaged1","plastic-damaged2")
```
```suggestion
	return list("plastic-damaged1", "plastic-damaged2")
```

Rohesie commented on code/game/turfs/open/floor/misc_floor.dm:
```diff
@@ -177,13 +181,17 @@
 	floor_tile = /obj/item/stack/tile/eighties
+
+/turf/open/floor/eighties/setup_broken_states()
 	broken_states = list("eighties_damaged")
```
```suggestion
	return list("eighties_damaged")
```

Rohesie commented on code/game/turfs/open/floor/plasteel_floor.dm:
```diff
@@ -1,7 +1,11 @@
 	floor_tile = /obj/item/stack/tile/plasteel
+
+/turf/open/floor/plasteel/setup_broken_states()
 	broken_states = list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
```
```suggestion
	return list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
```

Rohesie commented on code/game/turfs/open/floor/plasteel_floor.dm:
```diff
@@ -1,7 +1,11 @@
 	broken_states = list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5")
+
+/turf/open/floor/plasteel/setup_burnt_states()
 	burnt_states = list("floorscorched1", "floorscorched2")
```
```suggestion
	return list("floorscorched1", "floorscorched2")
```

Rohesie commented on code/game/turfs/open/floor/plating.dm:
```diff
@@ -20,6 +20,12 @@
 
 	var/attachment_holes = TRUE
 
+/turf/open/floor/plating/setup_broken_states()
+	broken_states = list("platingdmg1", "platingdmg2", "platingdmg3")
+
+/turf/open/floor/plating/setup_burnt_states()
+	burnt_states = list("panelscorched")
```
```suggestion
/turf/open/floor/plating/setup_broken_states()
	return list("platingdmg1", "platingdmg2", "platingdmg3")

/turf/open/floor/plating/setup_burnt_states()
	return list("panelscorched")
```

Rohesie commented on code/game/turfs/open/floor/plating/asteroid.dm:
```diff
@@ -24,6 +24,9 @@
 	var/postdig_icon_change = TRUE
 
+/turf/open/floor/plating/asteroid/setup_broken_states()
+	broken_states = list("asteroid_dug")
```
```suggestion
	return list("asteroid_dug")
```

Rohesie commented on code/game/turfs/open/floor/plating/asteroid.dm:
```diff
@@ -99,6 +102,9 @@
 	digResult = /obj/item/stack/ore/glass/basalt
 
+/turf/open/floor/plating/asteroid/basalt/setup_broken_states()
+	broken_states = list("basalt_dug")
```
```suggestion
	return list("basalt_dug")
```

Rohesie commented on code/game/turfs/open/floor/plating/asteroid.dm:
```diff
@@ -149,12 +155,13 @@
 	digResult = /obj/item/stack/sheet/mineral/snow
 
+/turf/open/floor/plating/asteroid/snow/setup_broken_states()
+	broken_states = list("snow_dug")
```
```suggestion
	return list("snow_dug")
```

Rohesie commented on code/game/turfs/open/floor/plating/asteroid.dm:
```diff
@@ -187,6 +194,8 @@
 	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
 
+/turf/open/floor/plating/asteroid/snow/ice/setup_broken_states()
+	broken_states = list("snow-ice")
```
```suggestion
	return list("snow-ice")
```

Rohesie commented on code/game/turfs/open/floor/plating/misc_plating.dm:
```diff
@@ -16,6 +16,9 @@
 	tiled_dirt = FALSE
 
+/turf/open/floor/plating/abductor/setup_broken_states()
+	broken_states = list("alienpod1")
```
```suggestion
	return list("alienpod1")
```

Rohesie reviewed (changes requested):
There a few more to go, but I guess you got the idea.
Make `setup_broken_states()` and `setup_brurnt_states()` return the list, instead of setting it up themselves.

AlinaStarkova committed:
changes how it sends the list

Rohesie commented on code/game/turfs/open/floor.dm:
```diff
@@ -28,22 +28,29 @@
-		broken_states = string_list(broken_states)
-	if(burnt_states)
-		burnt_states = string_list(burnt_states)
+		burnt_states = string_list(setup_burnt_states())
```
```suggestion
		var/list/new_burnt_states = setup_burnt_states()
		if(new_burnt_states)
			burnt_states = string_list(new_burnt_states)
```
This one was fine to keep. After all, there's no guarantee there will always be `burnt_states`

Rohesie commented on code/game/turfs/open/floor/mineral_floor.dm:
```diff
@@ -17,11 +17,12 @@
 	icons = typelist("icons", icons)
 
+/turf/open/floor/mineral/setup_broken_states()
+	if(!broken_states)
+		return list("[initial(icon_state)]_dam")
```
```suggestion
	return list("[initial(icon_state)]_dam")
```
This check is no longer relevant now.

AlinaStarkova committed:
prevent runtimes

tgstation-server commented:
Your Fix/Feature pull request delta is now above zero (4). Feel free to make Feature/Tweak/Balance PRs.
